### ISSUE TITLE
Build error during deployment

### ISSUE THREAD

**Issue by vtonu** [nuxt/nuxt#22769]
### Reproduction
You can find the repo on my GitHub(the nuxt-ui app) and try to deploy it yourself to see the error. (https://github.com/vtonu/Nuxt_UI)

### Describe the bug
npm run dev & build work fine in VSCode terminal without any issues. 

I've also deployed a similar page using react and deployment went smoothly. This is my first time trying to deploy a Vue/Nuxt app and I can't solve the issue. 

Clearing the cache didn't help nor deleting the package-lock.json help. (https://github.com/vuejs/vue-cli/issues/2404)
![image](https://github.com/nuxt/vercel-builder/assets/56773210/37c702cf-783e-4146-860c-03309cbbc57d)

![image](https://github.com/nuxt/vercel-builder/assets/56773210/6c20e5a0-415b-49bd-9e93-2fe3d659d95b)
npm run dev & npm run build work perfectly fine in VSCode.

### Additional context
I don't know why this won't deploy. https://github.com/nuxt/vercel-builder/issues/812

**Comment by danielroe** [comment]
Ahh, you're not using `nuxt/vercel-builder`.

I think the issue is likely that you have your project in a subdirectory and I suspect from your screenshot that the dependencies aren't being installed in your Vercel CI.

**Comment by vtonu** [comment]
> Ahh, you're not using `nuxt/vercel-builder`.
> 
> I think the issue is likely that you have your project in a subdirectory and I suspect from your screenshot that the dependencies aren't being installed in your Vercel CI.

That appears to be the case, but I had the exact same folder in a NextJS project (my-app folder) and it deployed fine. 

I took out everything from my-app folder, so it's inside Nuxt_UI folder itself, and it was deployed successfully. It also appears that now, on Vercel, when I go to import a repo from GitHub, it automatically sees that it's a NuxtJS app (logo)
![image](https://github.com/nuxt/nuxt/assets/56773210/04e19647-fab2-432a-9b41-ac377bdfcd69)
. Prior, it was unknown, and I had to manually set it to NuxtJS.

Not quite sure why in NextJS having a subdirectory (my-app) works fine, while in NuxtJS it doesn't.

Thank you for commenting on this, I would have given up on NuxtJS. :)
